src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.spec.ts

```diff
@@ -0,0 +1,52 @@
+import { DotPageSelectorService } from './dot-page-selector.service';
+import { DOTTestBed } from '../../../../../test/dot-test-bed';
+import { ConnectionBackend, ResponseOptions, Response } from '@angular/http';
+import { MockBackend } from '@angular/http/testing';
+import { mockPageSelector } from '../dot-page-selector.component.spec';
+
+describe('Service: DotPageSelector', () => {
+    beforeEach(() => {
+        this.injector = DOTTestBed.resolveAndCreate([DotPageSelectorService]);
+        this.dotPageSelectorService = this.injector.get(DotPageSelectorService);
+        this.backend = this.injector.get(ConnectionBackend) as MockBackend;
+        this.backend.connections.subscribe((connection: any) => (this.lastConnection = connection));
+    });
+
+    it('should get Page Selectors', () => {
+        let result;
+
+        this.dotPageSelectorService.getPagesInFolder('about').subscribe((res) => {
+            result = res;
+        });
+
+        this.lastConnection.mockRespond(
+            new Response(
+                new ResponseOptions({
+                    body: {
+                        contentlets: [mockPageSelector]
+                    }
+                })
+            )
+        );
+        expect(result).toEqual(mockPageSelector);
+    });
+
+    it('should get a specific Page Selector', () => {
```

Comment:
Is not getting an specific page selector, is getting a page by id.

Also need to test that the url and the body is being set to get the data is correct.